### PR TITLE
Add support for VFE Trenches

### PR DESCRIPTION
**Pull request by perkinslr**
## Additions

Describe new functionality added by your code, e.g.
- VFE Trenches adjust pawn heights

## Testing

Check tests you have performed:
- [X] Compiles without warnings
- [ ] Game runs without errors
- [ ] (For compatibility patches) ...with and without patched mod loaded
- [ ] Playtested a colony (specify how long)

### CODE REVIEW

**Review by N7Huntsman — DISMISSED** [review]
Appears to function in that pawns are harder to hit when in trenches, though I'm not entirely positive where their hitbox is actually at while in a trench. Their target height (as displayed when aiming at them) shifts from 1.05m in the open (aiming at the center of the torso) to 1.75m when in the trench (presumably aiming at the center of what's visible?).

File: Source/CombatExtended/CombatExtended/CollisionVertical.cs [hunk 1]
```diff
@@ -69,10 +70,12 @@ private static void CalculateHeightRange(Thing thing, out FloatRange heightRange
             
             float collisionHeight = 0f;
             float shotHeightOffset = 0;
+	    float heightAdjust = CETrenches.GetHeightAdjust(thing.Position, thing.Map);
```
Comment (MadaraUchiha) [inline Source/CombatExtended/CombatExtended/CollisionVertical.cs:73]: nitpick: inconsistent indent

File: Source/CombatExtended/Compatibility/Trenches.cs [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+	    vfeInstalled = ModLister.HasActiveModWithName(VFES_ModName);
+	}
+	public static float GetHeightAdjust(IntVec3 cell, Map map)
+	{
+	    if (cell == null || map == null)
+	    {
+		return 0;
+	    }
+	       
+	    if (vfeInstalled)
```
Comment (MadaraUchiha) [inline Source/CombatExtended/Compatibility/Trenches.cs:28]: Prefer early return

File: Source/CombatExtended/Compatibility/Trenches.cs [hunk 3]
```diff
@@ -0,0 +1,46 @@
+using System;
```
Comment (MadaraUchiha) [inline Source/CombatExtended/Compatibility/Trenches.cs:1]: Please format this file, the indentation here is weird

File: Source/CombatExtended/Compatibility/Trenches.cs [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+	    }
+	       
+	    if (vfeInstalled)
+	    {
+		
+		List<Thing> thingList = GridsUtility.GetThingList(cell, map);
+		foreach (Thing thing in thingList) 
+		{
+		    CompProperties_TerrainSetter compProperties = thing.def.GetCompProperties<CompProperties_TerrainSetter>();
+		    if (compProperties != null)
```
Comment (MadaraUchiha) [inline Source/CombatExtended/Compatibility/Trenches.cs:35]: Prefer early `continue`

**Review by MadaraUchiha — DISMISSED** [review]
Mostly formatting and code style nitpicks. Good to go with or without my comments :+1:

File: Source/CombatExtended/CombatExtended/CollisionVertical.cs (hunk 1 above)
Comment (perkinslr) [reply Source/CombatExtended/CombatExtended/CollisionVertical.cs:73]: Yeah, I think a format tool will be high on the list of things to do before 4.5...

File: Source/CombatExtended/Compatibility/Trenches.cs (hunk 2 above)
Comment (perkinslr) [reply Source/CombatExtended/Compatibility/Trenches.cs:28]: I considered that, but in this case if we decide to handle other mods, each one gets its own ` if (modXInstalled)`.

**Comment by perkinslr** [comment]
> Appears to function in that pawns are harder to hit when in trenches, though I'm not entirely positive where their hitbox is actually at while in a trench. Their target height (as displayed when aiming at them) shifts from 1.05m in the open (aiming at the center of the torso) to 1.75m when in the trench (presumably aiming at the center of what's visible?).

Is that a pawn aiming at them?  Or them aiming out?  I would expect them aiming out to go up, and a pawn aiming in to go down.

**Review by N7Huntsman — APPROVED** [review]

**Pull request merged**